File: android-lib/src/main/java/com/kinvey/android/callback/AsyncUploaderProgressListener.java
package com.kinvey.android.callback;


import com.kinvey.java.core.KinveyCancellableCallback;
import com.kinvey.java.core.MediaHttpUploader;
import com.kinvey.java.core.UploaderProgressListener;

import java.io.IOException;

public interface AsyncUploaderProgressListener<T> extends UploaderProgressListener, KinveyCancellableCallback<T> {
    @Override
    public void onSuccess(T result);

    @Override
    public void onFailure(Throwable error);

    @Override
    public void progressChanged(MediaHttpUploader uploader) throws IOException;

    @Override
    public void onCancelled();

    @Override
    public boolean isCancelled();
}

File: android-lib/src/main/java/com/kinvey/android/callback/AsyncUploaderProgressListener.kt
package com.kinvey.android.callback

import com.kinvey.java.core.KinveyCancellableCallback
import com.kinvey.java.core.MediaHttpUploader
import com.kinvey.java.core.UploaderProgressListener
import java.io.IOException

interface AsyncUploaderProgressListener<T> : UploaderProgressListener, KinveyCancellableCallback<T> {
    override fun onSuccess(result: T?)
    override fun onFailure(error: Throwable?)
    @Throws(IOException::class)
    override fun progressChanged(uploader: MediaHttpUploader?)

    override fun onCancelled()
    override var isCancelled: Boolean
}